owrx/source/__init__.py

```diff
@@ -589,7 +595,7 @@ def getDeviceOptionalKeys(self):
         return keys
 
     def getProfileMandatoryKeys(self):
-        return ["name", "center_freq", "samp_rate", "start_freq", "start_mod"]
+        return ["name", "center_freq", "samp_rate", "start_freq", "start_mod", "tuning_step"]
```

Comment:
There is no forcing going on here. It defaults to 1Hz, i.e. the original behavior. You can ask around though and easily find out that nobody needs to tune in 1Hz increments, even CW aficionados.

Reply:
You're forcing every receiver operator to fill this field. If the default is 1Hz, it can as well be absent and be added when necessary, thus avoiding unnecessary confusion.

Reply:
Since the field comes up with a default value, I am not forcing operator to do anything here. I am just making that field *visible* to the operator, which is a logical thing to do given that very, very few people expect to tune in 1Hz steps. Making visible is different from forcing, would you agree?

Reply:
Well then please explain to me why making the field visible is required. I still don't see why this should be a required field on all devices and profiles.

Also, making the field mandatory prevents the operator from using the inheritance (which would allow a default setting on the device, with the profiles automatically inheriting that setting).

Reply:
This field needs to be visible because in most cases the operator will want to set it. I do not know if you ever actually listen to a radio, but everyone who has knows that unless you tune to the exact frequency of the transmitter, your audio becomes garbled. And most transmitters (commercial or amateur) operate at certain steps, depending on the band and modulation. Tuning outside these steps, while occasionally useful, does not provide usable listening experience.

As to preventing inheritance, profiles already have the modulation setting that is also not present in the device settings. This does not prevent inheritance. The tuning step should be treated in pretty much the same way as modulation.

To make this argument more visual, I am going to include screenshots of several SDR packages for your viewing pleasure, showing tuning step right in front of the UI.

HDSDR:
![image](https://user-images.githubusercontent.com/3293202/180821935-24080b35-43a1-4c88-a6f7-8b48bb65903c.png)

SDR Console:
![image](https://user-images.githubusercontent.com/3293202/180822332-bd0c653b-3e8a-4a7f-9294-b48b94f026d2.png)

SDR Sharp:
![image](https://user-images.githubusercontent.com/3293202/180822545-04d4b0e7-f251-4dd3-8024-d47d68fd22d6.png)

Reply:
> This field needs to be visible because in most cases the operator will want to set it. I do not know if you ever actually listen to a radio, but everyone who has knows that unless you tune to the exact frequency of the transmitter, your audio becomes garbled. And most transmitters (commercial or amateur) operate at certain steps, depending on the band and modulation. Tuning outside these steps, while occasionally useful, does not provide usable listening experience.

This may be true for listening to broadcast stations, but this is not the only use case. When listening to ham broadcasts in SSB or CW you're better off with a more fine-grained control, and I don't think these modes profit from the tuning step at all. As such I don't agree with the opinion that operators will want to set this on every profile.

Also, please note that both transmitters as well as receivers may be off frequency (depending on their quality of calibration), so tuning to a presumably exact frequency may actually not yield the optimal result.

> As to preventing inheritance, profiles already have the modulation setting that is also not present in the device settings. This does not prevent inheritance. The tuning step should be treated in pretty much the same way as modulation.

I see, the setting is only available for profiles, so there's no inheritance, making this point moot. That then of course raises the question of whether operators could profit from enabling the inheritance.

> To make this argument more visual, I am going to include screenshots of several SDR packages for your viewing pleasure, showing tuning step right in front of the UI.

I'm sorry, but the fact that others do it some way has never been a good argument.

Reply:
_This may be true for listening to broadcast stations, but this is not the only use case. When listening to ham broadcasts in SSB or CW you're better off with a more fine-grained control, and I don't think these modes profit from the tuning step at all. As such I don't agree with the opinion that operators will want to set this on every profile._
SSB tuning is usually 500Hz, CW is 100Hz. Both tuning steps are present in my changes.

_I'm sorry, but the fact that others do it some way has never been a good argument._
Mmmm... ok. Then consider it a reason to _think_ what exactly makes all these others do it this way.

Reply:
Coming back to this, I think this discussion went in a completely wrong way. The question here should still be: is this a field that is required to achieve a working profile, and I'd say the answer to that is clearly: No.

The whole idea of having optional fields came from the original text-based configuration where the full extent of available keys could easily overwhelm operators. For that reason, many of the keys were optional and could be added to the configuration as needed after consulting the documentation.

This has been translated into the web configuration interface: It only shows the bare minimum, with the ability to add more functionality as needed. All the optional keys will have some kind of reasonable default.

This even applies to very common keys, for example "gain".

Reply:
For gain, you can safely leave it out and let it get derived from the device profile. The tuning step has nothing to do with the device profile and thus isn't present there. So, its default value cannot be derived from the device profile. It is supposed to be set based on the band frequency and modulation, i.e. specific to each band profile.

Since you are so eager to reduce the number of fields operator "needs" to enter, why do you have the starting frequency in the profile? What if I do not care about the starting frequency? Why not fix it to the center frequency by default, the same way you would like to keep the tuning step to 1Hz y default?

Reply:
I think the whole reduction of the configurable items is a topic of a separate discussion, since this dogmatic approach has ruined many products (GNOME, Windows 10, etc).

If you believe that a setting will be configured often, do not hide it. If you do not want to force the user to enter a value, provide a reasonable default value. This is exactly what has been done with the tuning step. In fact, I would insist on 1kHz as a reasonable default value, but to appease your sensibilities, it is set to 1Hz at the moment.

Reply:
I think that's the whole problem in this discussion: You expect this to be a feature that operators will want to use on all profiles, while I believe that many operators won't be bothered to set this up (correctly).

Reply:
Since there is a default value that retains the original behavior when not set, I see no problem with operators not bothering to set it.

Where I do see a problem is that tuning in 1Hz steps (the only option at the moment) makes precise tuning to a station nearly impossible without zooming the waterfall. And even with zoomed waterfall, it is often hit and miss. Thus is my conclusion that _most_ operators _will_ want to set the tuning step value.

Reply:
Your conclusion is not immediately wrong, but you're placing it into the wrong role. It's not the operators that want this feature, it's the users. This is confirmed by multiple requests on the project.

You're the first to involve the operator in this scenario, and while I do agree that it's not completely unreasonable to give the operator the ability to preselect a tuning step, it's important to understand that an operator should not be the one to make the final decision on this. As such, I find it's perfectly reasonable for an operator to skip this configuration and leave it completely up to the user.

Also, as I already stated previously, precision tuning is a myth. You will always have inaccuracies. You are using an rtl-sdr device yourself, right? It has a maximum precision of 1ppm, so even if you have done the calibration with utmost precision, you will still be up to 0.5 ppm off. Are you going to notice that? I doubt it.

Reply:
Look, I _can_ add this as a user-configurable option to the receiver panel, with the initial value configured by the operator (as done in the discussed change). The problem is that once I do this, you will reject the change claiming that I am cluttering the user interface. That is why I am not pushing any receiver panel changes, only sticking to safer, less visible operator interface changes.

Thus, for me, it is catch-22 situation. For you, it is "let the annoying freak run in circles, he is not getting in anyway". Alas, this is how it usually is with this kind of projects.